php/libraries/Utility.class.inc

```diff
@@ -1008,5 +1008,29 @@ class Utility
     {
         return rtrim($path, '/\\') . '/';
     }
-}
 
+    /**
+     * A PHP implementation of Go's filepath Join function.
+     *
+     * @param string ...$paths The path components to be joined.
+     *
+     * @return string Path components joined without any doubled forward-slashes.
+     *
+     * @see https://golang.org/pkg/path/filepath/#Join
+     */
+    public static function pathJoin(string ...$paths)
+    {
+        // This array stores the components of the final path.
+        $result = [];
+        foreach ($paths as $part) {
```

Comment:
This doesn't look like it'll properly handle paths with a leading "/". Can you add a test case for it?

Reply:
Yes, that's a great idea